internal/common/function/function.go

```diff
@@ -77,66 +77,125 @@ func ValidateBackupSpec(nonAdminBackup *nacv1alpha1.NonAdminBackup) error {
 	return nil
 }
 
-// GenerateVeleroBackupName generates a Velero backup name based on the provided namespace and NonAdminBackup name.
-// It calculates a hash of the NonAdminBackup name and combines it with the namespace and a prefix to create the Velero backup name.
-// If the resulting name exceeds the maximum Kubernetes name length, it truncates the namespace to fit within the limit.
-func GenerateVeleroBackupName(namespace, nabName string) string {
-	// Calculate a hash of the name
-	hasher := sha256.New()
-	_, err := hasher.Write([]byte(nabName))
-	if err != nil {
-		return ""
+// GenerateNacObjectNameWithUUID generates a unique name based on the provided namespace and object origin name.
+// It includes a UUID suffix. If the name exceeds the maximum length, it truncates nacName first, then namespace.
+func GenerateNacObjectNameWithUUID(namespace, nacName string) string {
+	// Generate UUID suffix
+	uuidSuffix := uuid.New().String()
```

Comment:
from function comment

> New creates a new random UUID or panics. New is equivalent to the expression

I would handle panic or at least add a TODO to do this later

Reply:
I can see the function may panic. This is only for the UUID4 implementation: 

https://github.com/google/uuid/blob/0e97ed3b537927cb4afea366bc4cc36f6eb37e75/version4.go#L9

As well in the doc generated from the above source code:
  https://pkg.go.dev/github.com/google/uuid#NewString

The Panic may originate from `Must` function and randomness source failure:
https://github.com/google/uuid/blob/0e97ed3b537927cb4afea366bc4cc36f6eb37e75/uuid.go#L45

Which is part of "crypto/rand".

It's highly unlikely the panic will ever occur and most likely it's caused by issues with the operating system or hardware (e.g., the random device /dev/urandom is unavailable or malfunctions).

I really don't know if we should be such detailed to handle this panic inside our code (we can by implementing defer function to handle panic). What will happen in unlikely event of panic is that  reconcile loop will crash and then it will be retried by the Kubernetes controller-runtime, so I think we are very safe here without additional wrappers around that case.